### PR TITLE
[Backend] Nostr publish

### PR DESCRIPTION
**Pull request by soosr**

### CODE REVIEW

File: WalletWasabi/Nostr/NostrExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+namespace WalletWasabi.Nostr;
+
+public record NostrCoordinator(string Description, string Name, Uri Uri, Network Network);
+
+public record GingerNostrCoordinator()
+	: NostrCoordinator(
+		Description: "Ginger Coordinator | FREE Remix, FREE under 0.01 BTC, FREE for Wasabi mixed coins | SAFE COINJOINS - Illicit actors are not allowed to participate!",
+		Name: "Ginger Coordinator",
+		Uri: new Uri("https://api.gingerwallet.io/"),
+		Network: Network.Main);
```
Comment (soosr) [inline WalletWasabi/Nostr/NostrExtensions.cs:19]: Please review the wording here. And also if it is ok to be hardcoded. @luzius1089 @CAnorbo

File: WalletWasabi.Backend/Global.cs [hunk 2]
```diff
@@ -33,10 +34,18 @@ public Global(string dataDir, IRPCClient rpcClient, Config config, IHttpClientFa
...
 
 		CoordinatorParameters = new(DataDir);
 		CoinJoinIdStore = CoinJoinIdStore.Create(CoordinatorParameters.CoinJoinIdStoreFilePath);
 
+		// Add Nostr publisher
+		// It is only supported on Mainnet.
+		if (Config.Network == Network.Main)
+		{
+			HostedServices.Register<CoordinatorNostrPublisher>(() => new CoordinatorNostrPublisher(TimeSpan.FromMinutes(15), NostrKeyManager.Key, new GingerNostrCoordinator()), "Coordinator Nostr Publisher");
+		}
```
Comment (soosr) [inline WalletWasabi.Backend/Global.cs:47]: Do not run this PR on mainnet, otherwise it will send the message out and it will appear to the public.

**Comment by soosr** [comment]
Ready for review! We agreed with @luzius1089 to only merge this after the initial testing of the backend.

**Comment by soosr** [comment]
@luzius1089 made it configurable so it only publishes if enabled. Ready to review and merge

File: WalletWasabi/Nostr/NostrExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+using NNostr.Client;
+using WalletWasabi.Exceptions;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Nostr;
+
+public record NostrCoordinator(string Description, string Name, Uri Uri, Network Network)
+{
+	private static NostrCoordinator Ginger = new(
+		Description: "Ginger Coordinator | FREE Remix, FREE under 0.01 BTC, FREE for Wasabi mixed coins | Secure Coinjoin - Illicit actors are not allowed to participate!",
```
Comment (soosr) [inline WalletWasabi/Nostr/NostrExtensions.cs:17]: Review the description please

Comment (luzius1089) [reply WalletWasabi/Nostr/NostrExtensions.cs:17]: how about: official coordinator, visit gingerwallet.io!

**Review by luzius1089 — CHANGES_REQUESTED** [review]

**Comment by luzius1089** [comment]
tACK

File: WalletWasabi/Nostr/NostrExtensions.cs (hunk 3 above)
Comment (soosr) [reply WalletWasabi/Nostr/NostrExtensions.cs:17]: This is how others advertise themself atm:

image

Comment (soosr) [reply WalletWasabi/Nostr/NostrExtensions.cs]: Pushed

**Pull request merged**